### PR TITLE
complete addition of TG_min_recap, and revise compreport

### PR DESCRIPTION
**Pull request by Rick-Methot-NOAA**
## What issue(s) does this PR address? Describe and add issue numbers, if applicable.
This adds capability to exclude from the tag logL tag groups with less than a specified minimum number of recaptures
- [#280  ] Link issue on pull request

## What tests have been done? Upload any model input files created for testing in a zip file, if possible.
works with the IOTC Yellowfin tuna spatial model
## What tests/review still need to be done? Who can do it, and by when is it needed (ideally)?
none
## Check which is true. This PR requires:

- [X] Changes in r4ss
- [X] Changes to SS3 manual
- [X] Changes to SSI (the SS3 GUI)
- [X] An entry in the stock synthesis change log (new features, bug reports)

##  Describe any changes in r4ss/SS3 manual/SSI/change log that are needed (if checked):
input of new feature is conditional on TG_mixperiod having a negative value:
1 # do tags (0/1)
121 # N tag groups
769 # N recap events
-4 # TG_mixperiod (latency) is the first period in which logL is calculated for this tag group (0 = release period)
#  (use negative value for TG_mixperiod to invoke reading TG_min_recap after TG_maxperiods
16 # max periods (seasons) to track recoveries, after which tags are still read into this accumulator period
2 # TG_min_recap:  N recaps >= mixperiod must be >= min_recap to include tag group in logL

## Additional information (optional):
@jyqalan

### CODE REVIEW

**Comment by k-doering-NOAA** [comment]
@Rick-Methot-NOAA, did you want this to be part of the 3.30.19?

**Comment by Rick-Methot-NOAA** [comment]
Yes because we still are waiting on the rebuilder revision.
Will need some work on the manual and r4ss.

File: SS_write_ssnew.tpl [hunk 1]
```diff
@@ -329,8 +329,15 @@ FUNCTION void write_nudata()
     // info on dimensions of tagging data
     report1<<N_TG<<" # N tag groups"<<endl;
     report1<<N_TG_recap<<" # N recap events"<<endl;
-    report1<<TG_mixperiod<<" # mixing latency period: N periods to delay before comparing observed to expected recoveries (0 = release period)"<<endl;
-    report1<<TG_maxperiods<<" # max periods (seasons) to track recoveries, after which tags enter accumulator"<<endl;
+    report1<<TG_mixperiod_rd<<" # TG_mixperiod (latency) is the first period in which logL is calculated for this tag group (0 = release period)"<<endl;
+    report1<<"#  (use negative value for TG_mixperiod to invoke reading TG_min_recap after TG_maxperiods"<<endl;
+    report1<<TG_maxperiods<<" # max periods (seasons) to track recoveries, after which tags are still read into this accumulator period"<<endl;
+    if(TG_mixperiod_rd>=0){
+      report1<<"#  TG_min_recap would go here if TG_mixperiod < 0"<<endl;
```
Comment (k-doering-NOAA) [inline SS_write_ssnew.tpl:336]: Maybe reword to `"# COND: TG_min_recap if TG_mixperiod < 0"` to be more consistent in style with the manual.

File: SS_write_ssnew.tpl [hunk 2]
```diff
@@ -329,8 +329,15 @@ FUNCTION void write_nudata()
...
+    }
+    else{
+      report1<<TG_min_recap<<" # TG_min_recap:  N recaps >= mixperiod must be >= min_recap to include tag group in logL"<<endl;
```
Comment (k-doering-NOAA) [inline SS_write_ssnew.tpl:339]: Is there a way to reword this more concisely? I am having trouble following the statements. Maybe just simpl put 
`# COND: TG_min_recap if TG_mixperiod < 0` to mirror the statement produced when the mix period is greater than 0?

**Review by k-doering-NOAA — APPROVED** [review]
Looks good! I took a look at the code, then Rick walked me through the changes while screen sharing and explained why they were made. Makes sense to me.

Optional changes:
- ssnew comments for the new conditional input
- using a new 2 option for the do tag flag as the trigger for the conditional input rather than a negative value on the mix period. Rick made the observation that this may be better and I agree it may be more intuitive for users.

**Comment by Rick-Methot-NOAA** [comment]
revise trigger
2 # do tags (0/1/2); where 2 allows entry of TG_min_recap
instead of negative mixperiod value as trigger.

**Comment by k-doering-NOAA** [comment]
Looks good @Rick-Methot-NOAA ! I ran a version of the YFT for the spatial project, and it worked well. Attached are the input files for the version with the min recapture option for future.

Running with `-stopph 0 -nohess`, only the 2 tag likelihoods changed when using the min recapture option, and the likelihoods were reduced, so this checks out.

I also tried running with estimation as well, and the model ran fine.

Something picky: I noticed there was a space in the .ss_new file when printing the # COND line; I will push a small change to get rid of it. 

This is ready to merge in once I push the picky change.

[use_min_option_inputfiles.zip](https://github.com/nmfs-stock-synthesis/stock-synthesis/files/8389962/use_min_option_inputfiles.zip)
